xapian-core/net/remote_protocol.rst

```diff
@@ -318,3 +318,33 @@ Reconstruct text
 
 -  ``MSG_RECONSTRUCTTEXT I<did> I<length> I<start_pos> I<end_pos> <prefix>``
 -  ``REPLY_RECONSTRUCTTEXT <text>``
+
+Synonym Term List
+-----------------
+
+-  ``MSG_SYNONYMTERMLIST <word>``
+-  ``REPLY_SYNONYMTERMLIST [C<chars of previous term to reuse> S<string to append>]...``
+
+Synonym Key List
+-----------------
+
+-  ``MSG_SYNONYMKEYLIST <word>``
+-  ``REPLY_SYNONYMKEYLIST [C<chars of previous term to reuse> S<string to append>]...``
+
+Add synonym
+-----------
+
+- ``MSG_ADDSYNONYM <word> <synonym>
```

Comment:
This is wrong - `pack_string()` is used on `word` so it's `S<word> <synonym>`.  Same for `MSG_REMOVESYNONYM`.